src/content/docs/en/guides/backend/google-firebase.mdx

```diff
@@ -11,6 +11,537 @@ import FileTree from '~/components/FileTree.astro'
+
+Next, we will edit the `signin.astro` page to add a button to sign in with Google and a script to handle the sign in process.
+
+```astro title="src/pages/signin.astro" ins={27, 34, 35, 65-78}
+---
+import { app } from "../firebase/server";
+import { getAuth } from "firebase-admin/auth";
+import Layout from "../layouts/Layout.astro";
+
+/* Check if the user is authenticated */
+const auth = getAuth(app);
+const sessionCookie = Astro.cookies.get("session").value;
+if (sessionCookie) {
+  const decodedCookie = await auth.verifySessionCookie(sessionCookie);
+  if (decodedCookie) {
+    return Astro.redirect("/dashboard");
+  }
+}
+---
+
+<Layout title="Sign in">
+	<h1>Sign in</h1>
+  <p>New here? <a href="/register">Create an account</a></p>
+	<form action="/api/auth/signin" method="post">
+    <label for="email" for="email">Email</label>
+		<input type="email" name="email" id="email" />
+    <label for="password">Password</label>
+		<input type="password" name="password" id="password" />
+		<button type="submit">Login</button>
+	</form>
+  <button id="google">Sign in with Google</button>
+</Layout>
+<script>
+  import {
+    getAuth,
+    inMemoryPersistence,
+    signInWithEmailAndPassword,
+    GoogleAuthProvider,
+    signInWithPopup,
+  } from "firebase/auth";
+  import { app } from "../firebase/client"
+
+  const auth = getAuth(app);
+  auth.setPersistence(inMemoryPersistence)
+
+  const form = document.querySelector('form') as HTMLFormElement;
+  form.addEventListener('submit', async (e) => {
+    e.preventDefault();
+    const formData = new FormData(form);
+    const email = formData.get('email')?.toString();
+    const password = formData.get('password')?.toString();
+
+    if (!email || !password) {
+      return;
+    }
+    const userCredential = await signInWithEmailAndPassword(auth, email, password)
+    const idToken = await userCredential.user.getIdToken();
+    const response = await fetch('/api/auth/signin', {å
+      headers: {
+        'Authorization': `Bearer ${idToken}`
+      }
+    })
+    
+    if (response.redirected) {
+      window.location.assign(response.url)
+    }
+  });
+
+  const googleSignin = document.querySelector('#google') as HTMLButtonElement;
+  googleSignin.addEventListener("click", async () => {
+    const provider = new GoogleAuthProvider();
+    const userCredential = await signInWithPopup(auth, provider);
+    const idToken = await userCredential.user.getIdToken();
+    const res = await fetch("/api/auth/signin", {
+    headers: {
+      Authorization: `Bearer ${idToken}`,
+    },
+  });
+
+  if (res.redirected) {
+    window.location.assign(res.url);
+  }})
+</script>
+```
+
+The google sign in button will open a popup window to sign in with Google. Once the user signs in, it will send a `POST` request to the `/api/auth/signin` endpoint with the ID token generated by OAuth provider. 
```

Comment:
```suggestion
When clicked, the Google sign in button will open a popup window to sign in with Google. Once the user signs in, it will send a `POST` request to the `/api/auth/signin` endpoint with the ID token generated by OAuth provider. 
```